attic/data_backend_forecasting.md

```diff
@@ -0,0 +1,234 @@
+different data types.
+
+Used formats: 
+- Basic: `ts`: Example `r data(Airpassengers)`
+- Advanced: `xts`: [Cheatsheet](https://www.datacamp.com/community/blog/r-xts-cheat-sheet) from library(xts)
+- More general: `prophet` expects a `data.frame` with a `ds` (`as.Date`) and `y` (value) column.
+- Very important: How was this solved in the old `mlr`?
+
+Way more other formats exist, there is also a package that converts time-series formats around:
+[tsbox](https://github.com/christophsax/tsbox)
+This has another helpfull resource: [link](https://github.com/christophsax/tsbox)
+
+## Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Different forecasting methods require the data to be stored in different formats.
+It is imperative to store data in a format that is flexible enough for different scenarios:
+
+1. one or more time-series
+2. time-series can be equi-distant or  not
+3. time-series can be measured at equivalent grids (or not).
+
+
+Example: 
+
+```r
+# Equi-distant and same grid:
+ts1 = ts(rnorm(12), frequency = 4, start = 2000)
+ts2 = ts(rnorm(12), frequency = 4, start = 2000)
+# Different grid
+ts1 = ts(rnorm(12), frequency = 4, start = 2000)
+ts2 = ts(rnorm(36), frequency = 12, start = 2000)
+```
+
+
+Proposals:
+
+(Long Format): From `tsbox`: 
+
+```
+ In data frames, multiple time series will be stored in a 'long'
+     format. tsbox detects a _value_, a _time_ and zero to several _id_
+     columns. Column detection is done in the following order:
+
+       1. Starting *on the right*, the first first ‘numeric’ or
+          ‘integer’ column is used as *value column*.
+
+       2. Using the remaining columns, and starting on the right again,
+          the first ‘Date’, ‘POSIXct’, ‘numeric’ or ‘character’ column
+          is used as *time column*. ‘character’ strings are parsed by
+          ‘anytime::anytime()’. The time stamp, ‘time’, indicates the
+          beginning of a period.
+
+       3. *All remaining* columns are *id columns*. Each unique
+          combination of id columns points to a time series.
+
+```
+This would look as follows (output is a data.table):
+```
+> ts_dt(x.ts)
+          id       time value
+  1: mdeaths 1974-01-01  2134
+  2: mdeaths 1974-02-01  1863
+  3: mdeaths 1974-03-01  1877
+  4: mdeaths 1974-04-01  1877
+  5: mdeaths 1974-05-01  1492
+ ---                         
+140: fdeaths 1979-08-01   379
+141: fdeaths 1979-09-01   393
+142: fdeaths 1979-10-01   411
+143: fdeaths 1979-11-01   487
+144: fdeaths 1979-12-01   574
+```
+
+
+## Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+@mllg created a draft for a `DataBackend`. This can serve as basis for discussion.
+
+
+```r
+DataBackendLong = R6Class("DataBackendLong",
+  inherit = mlr3::DataBackend,
+  cloneable = FALSE,
+
+  public = list(
+    id_col = NULL,
+    value_col = NULL,
+
+    initialize = function(data, primary_key, id_col) {
```

Comment:
The primary key is the time column, the id column is equivalent to the `id` column in the tsbox example above